### DISCUSSION TITLE
git lfs installed, git checkout some files only downloads pointers on debian 11 but in windows works ok

### DISCUSSION THREAD

**Discussion by Davidn0** [git-lfs/git-lfs#5335] [General] [answered]
Git lfs is installed, but git checkout branch some files only downloads pointers on debian 11, i need to do git lfs pull to download the binaries and then it shows as modified files, it takes the pointer files as unchanged.
But on windows only git checkout branch already downloads the lfs files binaries and git status says "nothing to commit, clean working tree" , it's works well

What am I doing wrong?

git lfs env windows:
git-lfs/3.3.0 (GitHub; windows amd64; go 1.19.3; git 77deabdf)
git version 2.40.0.windows.1

git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"

git lfs env debian:
git-lfs/3.3.0 (GitHub; linux amd64; go 1.19.3)
git version 2.30.2

git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"

**Comment by Davidn0** [comment, marked as answer]
I think I found my problem, it turns out that it's not all files that don't convert to binary automatically, some do it correctly. The problem seems to be that some files have the uppercase extension and these are the ones left as pointers if I don't do git lfs pull

How should I proceed, changing gitattributes? Can I make it case insensitive?

**Reply by bk2204** [reply]
On Linux, Git LFS is always case sensitive because Linux file systems are generally case sensitive.  If you want the files with the uppercase extension to be LFS files, then you need to add them to `.gitattributes` with that extension.  For example, if you have this:

```
*.png filter=lfs diff=lfs merge=lfs !text
```

then you'll also need this:

```
*.PNG filter=lfs diff=lfs merge=lfs !text
```

**Reply by Davidn0** [reply]
Ok, thanks for the quick response and for explaining the reason.

I will make the changes

**Comment by Davidn0** [comment]
I have already tried everything and it works correctly after adding the uppercase extensions

Thanks, @bk2204 

In case it helps someone use this to speed it up:
cat .gitattributes | sed -n -e 's/[\*\.][[:alnum:]]*/\U&/gp'
and then add them
